src/store/zoom.ts

```diff
@@ -0,0 +1,96 @@
+import { createSlice, PayloadAction, createSelector } from "@reduxjs/toolkit";
+
+interface ZoomState {
+  byId: {
+    [id: string]: {
+      id: string;
+      name: string;
+    };
+  };
+  allIds: string[];
+  online: string[];
```

Comment:
I think this state could be cleaned up to remote a) duplication, and b) the opportunity for corruption/inconsistency.

Some thoughts, roughly in order of importance:
1. `allIds` shouldn't exist, just use `byId.keys()`/`Object.keys(byId)`.
2. I don't see a compelling reason for the link slice to have a separate `byId` object. I think this schema should just get a `discordId` property added to it (and, consequently, the current `id` property should be renamed to `zoomId`).
3. Maybe consider making `byId` a `Map`, to get proper support for things like `has`, etc. Also prevents a theoretical (but probably not that problematic) situation where someone inputs `__proto__` or something.

Reply:
1. Agreed
2. Agreed (at some point I was handling links different and it was neccessary, but not it's not)
3. See other comment about maps and redux/immer. All the ids are coming from discord/zoom, not user input, so we shouldn't be able to have issues with `__proto__` or malicious input